Added Docker scripts in the new Docker directory

jhsu802701 opened the pull request:
#### What? Why?
These changes make it so much easier to use Docker.  You no longer have to keep typing "docker-compose run --rm web".

#### What should we test?
This pull request adds scripts in the "docker" directory and updates the documentation.  The mechanisms behind the app are unchanged.

#### Release notes
Added scripts to make it easier to use Docker.  You no longer have to keep typing "docker-compose run --rm web".

#### Documentation updates
The URL leading to the Docker page should be updated.

luisramos0 reviewed (commented):
This looks very good, thank you!
I want to give this a try before I approve.

I think that at the end of the DOCKER.md we could have a bullet list of the scripts available and what each of them does in very few words.

jhsu802701 commented:
I added a summary of the Docker scripts to the end of the docker/README.md file (which replaces DOCKER.md) .  This pull request already includes an updated link to the Docker page in GETTING_STARTED.md.

luisramos0 commented on docker/README.md:
```diff
@@ -0,0 +1,54 @@
+* docker/server: Use this script to run this app in the Rails server.  This script executes the "docker-compose up" command and logs the results.  If all goes well, you will be able to view this app on your local browser at http://localhost:3000/.
+* docker/test: Use this script to run the entire test suite.
+* docker/testc: Use this script to run just the controller tests.
+* docker/testm: Use this script to run just the model tests.
```
nice!
can we order this in the order of typical usage?
- build
- server
- test
- exec
- seed
- nuke

luisramos0 reviewed (approved):
Nice :ok_hand:

I think I'd remove testc and testm, there are too slow. devs will always run specific specs when they are preparing PRs.

jhsu802701 commented:
I removed docker/testc and docker/testm and made your requested changes to the order of the scripts listed in the summary section.

luisramos0 reviewed (approved):
:ok_hand:

jhsu802701 commented:
Is there anything else I should add, remove, or change?

sigmundpetersen commented:
Hey @jhsu802701 , this just needs an approval from another reviewer and will then move on to QA 👍

sauloperez reviewed (dismissed):
The only bit from the decorator I couldn't find is https://github.com/openfoodfoundation/openfoodnetwork/blob/427d1aaeff51234b8995351f555f1cfd7bb9bb2d/app/models/spree/order_decorator.rb#L32-L40. Where did that go?

sauloperez commented on docker/cop:
```diff
@@ -0,0 +1,11 @@
+echo '----------------------------------------------------'
+echo 'BEGIN: docker-compose run web bundle exec rubocop -D'
+echo '----------------------------------------------------'
+docker-compose run web bundle exec rubocop -D
```
we're missing `--rm` after `run`, right?

sauloperez replied:
btw, -D seems to be the default already

```
-D, --[no-]display-cop-names     Display cop names in offense messages.
                                                   Default is true.
```

sauloperez commented on docker/nukec:
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# This script destroys all Docker containers and networks but leaves the Docker images alone.
```
it'll destroy all running containers, right? I'd clarify that in that case.

sauloperez commented on docker/nukec:
```diff
@@ -0,0 +1,19 @@
+
+echo '-----------------------'
+echo 'docker network prune -f'
+docker network prune -f
```
but that will destroy all networks, including any other not related to OFN?

sauloperez commented on docker/seed:
```diff
@@ -0,0 +1,27 @@
+echo '--------------------------------------------------------------'
+echo 'BEGIN: docker-compose run web bundle exec rake ofn:sample_data'
+echo '--------------------------------------------------------------'
+docker-compose run web bundle exec rake ofn:sample_data
```
aren't we missing `--rm` here too? why don't reuse the script you added?

sauloperez commented on docker/exec:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
what concerns me a bit of this command is that it mixes the meanings of docker-compose's exec and run. Do you see any problem renaming it to `docker/run` so we stick to that meaning?

jhsu802701 commented:
- I renamed docker/exec to docker/run and updated the docker/README.md file to reflect this.

- I moved the "docker prune network -f" command from docker/nukec to docker/nuke.

- I corrected the "docker-compose" commands in the scripts in the Docker directory.

- I updated the docker/cop script to remove the "-D" flag (due to the redundancy).

- The contents of the DOCKER.md file (including the section on troubleshooting Windows) have been moved to the docker/README.md file.

jhsu802701 commented:
Are there any additional changes I need to make?  Until this pull request is merged, people will have to keep typing "docker-compose run --rm web".

sauloperez reviewed (approved):
I just took a look and it works. Thanks for this contribution @jhsu802701 !
